core/utils/ethabi.go

```diff
@@ -48,13 +49,15 @@ func EVMTranscodeBytes(value gjson.Result) ([]byte, error) {
 		return EVMEncodeBytes(EVMWordUint64(1))
 
 	case gjson.Number:
-		word, err := EVMWordSignedBigInt(big.NewInt(int64(value.Num)))
+		v := big.NewFloat(value.Num)
+		if !v.IsInt() {
+		}
+		vInt, _ := v.Int(big.NewInt(0))
+		word, err := EVMWordSignedBigInt(vInt)
```

Comment:
Wouldn't this always be 0?

Reply:
You'd think so, but this is part of the `big` package's API, which has a lot of awkwardnesses like this. Here is the `big.Float.Int` doc string:

```go
// Int returns the result of truncating x towards zero;
//                                    ^^^^^
// or nil if x is an infinity.
// The result is Exact if x.IsInt(); otherwise it is Below
// for x > 0, and Above for x < 0.
// If a non-nil *Int argument z is provided, Int stores
// the result in z instead of allocating a new Int.
func (x *Float) Int(z *Int) (*Int, Accuracy) {
    ^^^^
```

The `big.NewInt(0)` is just a way to allocate space for the `big.Int` answer. I'll make it clearer.